tests/test_monotonic_constraints.py

```
@@ -0,0 +1,1287 @@
+        # Layout: [num0, cat0, num1, cat1, num2]
+        # Global indices: num0=0, cat0=1, num1=2, cat1=3, num2=4
+        np.random.seed(42)
+        n_samples = 300
+        
+        # Numerical features
+        num0 = np.random.randn(n_samples).astype(np.float32)
+        num1 = np.random.randn(n_samples).astype(np.float32)
+        num2 = np.random.randn(n_samples).astype(np.float32)
+        
+        # Categorical features
+        cat0 = np.random.choice(['A', 'B', 'C'], n_samples)
```

Ruff (0.14.14)

[error] Local variable `cat0` is assigned to but never used

Remove assignment to unused variable `cat0`

(F841)